### PR TITLE
Ensuring that a filter is specified as a query parameter only once in the curl URL for CSS

### PR DESCRIPTION
**Pull request by poojamaiya**
#### Summary
Ensuring that a filter is specified as a query parameter only once in the curl URL for CSS. Addresses issue #131

#### Implementation details
Parsing through the filter map request URL query to make sure that there is only one value per key (i.e. filter) 

#### Testing

- [X] cluster-state-service binary built locally and unit-tests pass (`cd cluster-state-service; make; cd ../`)
- [X] cluster-state-service build in Docker succeeds (`cd cluster-state-service; make release; cd ../`)
- [X] Manually tested using the following curl commands
* curl -v "http://localhost:3000/v1/instances?cluster=cluster1&cluster=cluster2"  - Errors out with message At least one of the filters provided is specified multiple times`
* curl -v "http://localhost:3000/v1/tasks?cluster=cluster1&cluster=cluster2"  - Errors out with message At least one of the filters provided is specified multiple times`

New tests cover the changes: Added unit tests. Added e2e tests using http calls.

#### Description for the changelog
* Making sure that a filter is specified only once when calling the List API using curl

#### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

#### Before merging

- [X] cluster-state-service end-to-end tests pass. Required setup details are listed [here](https://github.com/blox/blox/blob/dev/cluster-state-service/internal/Readme.md).

### CODE REVIEW

File: cluster-state-service/internal/features/e2e_list_tasks.feature [hunk 1]
```diff
@@ -64,3 +64,8 @@ Feature: List Tasks
     When I try to list tasks with an invalid cluster filter
     Then I get a ListTasksBadRequest task exception
     And the task exception message contains "Invalid cluster ARN or name"
+
+  Scenario: List instances with redundant filters
```
Comment (narehayrapetyan) [inline cluster-state-service/internal/features/e2e_list_tasks.feature]: List instances -> List tasks

File: cluster-state-service/internal/features/e2einstancesteps/list_instances_steps.go [hunk 2]
```diff
@@ -94,4 +100,28 @@ func init() {
...
+			T.Errorf("Unknown exception type '%s' when trying to list instances with redundant filters", resp.Status)
+		}
+
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			T.Errorf("Error reading expection message when trying to list instances with redundant filters")
+		}
+		exceptionMsg := string(body)
+		exceptionList = append(exceptionList, Exception{exceptionType: exceptionType, exceptionMsg: exceptionMsg})
```
Comment (narehayrapetyan) [inline cluster-state-service/internal/features/e2einstancesteps/list_instances_steps.go]: is the exception message verified somewhere else?

File: cluster-state-service/internal/features/e2e_list_tasks.feature (hunk 1 above)
Comment (poojamaiya) [inline cluster-state-service/internal/features/e2e_list_tasks.feature]: Whoops! thanks, fixing.

File: cluster-state-service/internal/features/e2einstancesteps/list_instances_steps.go (hunk 2 above)
Comment (poojamaiya) [inline cluster-state-service/internal/features/e2einstancesteps/list_instances_steps.go]: Yup. Its being verified in one of the subsequent steps - "And the instance exception message contains "At least one of the filters provided is specified multiple times"

**Pull request merged**